python/lsst/daf/butler/core/config.py

```diff
@@ -70,6 +70,36 @@ def _doUpdate(d, u):
     return d
 
 
+def _checkNextItem(k, d, create, must_be_dict):
+    """See if k is in d and if it is return the new child."""
+    nextVal = None
+    isThere = False
+    if d is None:
+        # We have gone past the end of the hierarchy
+        pass
+    elif not must_be_dict and isinstance(d, collections.abc.Sequence):
+        # Check sequence first because for lists
+        # __contains__ checks whether value is found in list
+        # not whether the index exists in list. When we traverse
+        # the hierarchy we are interested in the index.
+        try:
+            nextVal = d[int(k)]
+            isThere = True
+        except IndexError:
+            pass
```

Comment:
I'm not familiar with all of the Config code.  With that in mind, can d be a list of numbers (d = [10,20,30])?  If yes (list is an instance of collections.abc.Sequence), and if k is 30, then it will get an IndexError and isThere will be False, but based on the ValueError line below I think we're wanting isThere to be True.

Reply:
No. It's something like `config[".a.2.b"]` and `a` refers to a sequence, `2` to index 2 and then `b` is a key in the dict inside that sequence.  It's features like this that are great for one off lookups in a config but a disaster when doing many of them because you get the isinstance check for sequence every time (except now I've bypassed it for a non-hierarchical look up since it's impossible for a Config top level to be a sequence.